### ISSUE TITLE
False positive: Unused local variable $cs

### ISSUE THREAD

**Issue by machour** [kalessil/phpinspectionsea#724]
```php 
$cs = $rs = 1;
if (isset($attr['COLSPAN']) && $attr['COLSPAN'] > 1) {
     // here the Unused local variable is raised for $cs
     $cs = $this->mpdf->cell[$this->mpdf->row][$this->mpdf->col]['colspan'] = $attr['COLSPAN'];
}
// but it's used in this if condition
if ($this->mpdf->table[$this->mpdf->tableLevel][$this->mpdf->tbctr[$this->mpdf->tableLevel]]['nc'] < $this->mpdf->col + $cs) {
		// ...
```

**Comment by rentalhost** [comment]
**Simplified case:**

```php
class Example
{
    public function test()
    {
        // ▼ marked as unused incorrectly.
        $used = $this->someArray[1][1] = 'any value kind';
        
        // If you just remove one of [1] key access, it will works.
        // Ex.: $used = $this->someArray[1] = 'any value kind;

        return $used;
    }
}
```

The problem occurs when you access a _key-of-key-of-property_ (`$this->someArray[1][1]`) and set two variables once. For some reason, the `$user` pass to be marked as unused.

**Comment by kalessil** [comment]
Thank you for reporting @machour, unfortunately, this false-positive is by PhpUnusedLocalVariableInspection. Please report to PhpStorm team.

**Comment by machour** [comment]
Oh sorry! 🙈 
How can I distinguish between PhpStorm's inspections and yours?

**Comment by kalessil** [comment]
No problems =)
Try suppressing a warning (e.g. `/** @noinspection ReturnTypeCanBeDeclaredInspection */` will be added) and just search `ReturnTypeCanBeDeclaredInspection` in our repo.
